go/services/reports/internal/services/reports_service.go

```diff
@@ -145,6 +113,124 @@ func (s *ReportsService) ScheduleReport(
 	return report, nil
 }
 
+func (s *ReportsService) ScheduleNodeInsights(
+	ctx context.Context,
+	params ReportGenerationFilters) (
+	*insights.ScheduledNodeInsights,
+	int, error) {
+
+	sinceDate := time.UnixMilli(params.SinceMs)
+	toDate := time.UnixMilli(params.ToMs)
+
+	nodeLogs, err := s.GetNodeLogsByParams(
+		ctx,
+		params.ClusterId,
+		sinceDate,
+		toDate)
+	if err != nil {
+		s.logger.Error("Failed to fetch application logs", zap.Error(err))
+		return nil, 0, err
+	}
+
+	var aggregatedNodeInsights = insights.ScheduledNodeInsights{}
+	analyzedNodeEntries := 0
+
+	for {
+		if !nodeLogs.HasNextBatch() {
+			break
+		}
+
+		nextLogsBatch, err := nodeLogs.GetNextBatch()
+		if err != nil {
+			s.logger.Error("Failed to get next batch of logs")
+		}
+
+		groupedLogs := sharedrepositories.GroupNodeLogsByName(nextLogsBatch)
+		configurationsByNode := insights.MapNodeNameToConfiguration(params.NodeConfiguration)
+
+		// In place filter based on node configuration
+		filter.FilterByNodesAccuracy(groupedLogs, configurationsByNode)
+
+		analyzedNodeEntries += filter.CountFilterResultEntries(groupedLogs)
+
+		nodeInsights, err := s.nodeInsightsGenerator.ScheduleNodeInsights(
+			groupedLogs,
+			configurationsByNode,
+			params.ClusterId,
+			params.SinceMs,
+			params.ToMs,
+		)
+
+		aggregatedNodeInsights = insights.ScheduledNodeInsights{
+			ScheduledJobIds:   append(aggregatedNodeInsights.ScheduledJobIds, nodeInsights.ScheduledJobIds...),
+			SinceMs:           nodeInsights.SinceMs,
+			ToMs:              nodeInsights.ToMs,
+			ClusterId:         nodeInsights.ClusterId,
+			NodeConfiguration: nodeInsights.NodeConfiguration,
+		}
+	}
+	return &aggregatedNodeInsights, analyzedNodeEntries, nil
+}
+
+func (s *ReportsService) ScheduleApplicationInsights(
+	ctx context.Context,
+	params ReportGenerationFilters) (
+	*insights.ScheduledApplicationInsights,
+	int, error) {
+
+	sinceDate := time.UnixMilli(params.SinceMs)
```

Comment:
Just a NIT, a single var() block with var declarations in the beginning of a function can improve readability

Reply:
I agree